includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1371,13 +1400,35 @@ private function get_payment_methods_from_request() {
 			$payment_methods = [ str_replace( 'woocommerce_payments_', '', $upe_payment_method ) ];
 		} elseif ( WC_Payments_Features::is_upe_split_enabled() || WC_Payments_Features::is_upe_deferred_intent_enabled() ) {
 			$payment_methods = [ 'card' ];
+			if ( WC_Payments_Features::is_upe_deferred_intent_enabled() &&
+				in_array( Payment_Method::CARD, $payment_methods, true ) &&
+				in_array( Payment_Method::LINK, $this->get_upe_enabled_payment_method_ids(), true ) ) {
+				$payment_methods[] = Payment_Method::LINK;
+			}
 		} else {
 			$payment_methods = WC_Payments::get_gateway()->get_payment_method_ids_enabled_at_checkout( null, true );
 		}
 
 		return $payment_methods;
 	}
 
+	/**
+	 * Get values for Stripe mandate_data parameter
+	 *
+	 * @return array mandate_data values to use in request.
+	 */
+	private function get_mandate_data() {
+		return [
+			'customer_acceptance' => [
+				'type'   => 'online',
+				'online' => [
+					'ip_address' => WC_Geolocation::get_ip_address(),
+					'user_agent' => 'WooCommerce Payments/' . WCPAY_VERSION_NUMBER . '; ' . get_bloginfo( 'url' ),
```

Comment:
Our usage of user agent parameters is pretty inconsistent. Our API client uses `"WooCommerce Payments/" + WCPAY_VERSION_NUMBER`, but there are a few other places where the more specific blog URL is also added. I'd think either that or the Stripe account ID would probably be most relevant here, so included the former. Also, not sure if there's a better way to collect the IP address without the WC dependency. 🤷